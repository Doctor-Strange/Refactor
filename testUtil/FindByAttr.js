

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
