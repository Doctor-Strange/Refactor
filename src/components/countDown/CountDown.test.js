import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../testUtil/FindByAttr";
import CountDown from "./";

const setup = (props) => {
  return mount(<CountDown {...props} />);
};

const props = {
  time: 20,
  Done: () => {},
};

describe("CountDown component", () => {
  describe("snapshot", () => {
    test("/components/calculator", () => {
      const wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe("Time props", () => {
    test("set Timer", () => {
      const wrapper = setup(props);
      const time = FindByAttr(wrapper, "time");
      expect(time.text()).toEqual("20");
    });
  });
});
