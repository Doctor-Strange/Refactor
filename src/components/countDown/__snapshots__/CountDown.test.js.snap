// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDown component snapshot /components/calculator 1`] = `ReactWrapper {}`;
